flyway-core/src/main/java/org/flywaydb/core/internal/dbsupport/saphana/SapHanaSqlStatementBuilder.java

```diff
@@ -44,22 +40,32 @@ protected String cleanToken(String token) {
 
     @Override
     protected Delimiter changeDelimiterIfNecessary(String line, Delimiter delimiter) {
-        if (StringUtils.countOccurrencesOf(statementStart, " ") < 4) {
```

Comment:
I found the approach useful to determine the start of the statement we're building. But it only worked in very limited cases. I tried to extend it by ignoring multiple blank lines, line comments and the beginning of block comments.

I also added tests specifically for the case you mentioned.